packages/ses/src/lockdown.js

```diff
@@ -100,10 +100,30 @@ const safeHarden = makeHardener();
 // only ever need to be called once and that simplifying lockdown will improve
 // the quality of audits.
 
-const assertDirectEvalAvailable = () => {
-  let allowed = false;
+const probeHostEvaluators = () => {
+  let functionAllowed;
   try {
-    allowed = FERAL_FUNCTION(
+    functionAllowed = FERAL_FUNCTION('return true')();
+  } catch (_error) {
+    // We reach here if the Function() constructor has not been implemented by the
+    // host, or is outright forbidden by a Content Security Policy.
+    functionAllowed = false;
+  }
+
+  let evalAllowed;
+  try {
+    evalAllowed = FERAL_EVAL('true');
+  } catch (_error) {
+    // We reach here if eval() has not been implemented by the host, or is outright
+    // forbidden by a Content Security Policy.
```

Comment:
```suggestion
    // We reach here if eval() has not been implemented by the host or is forbidden,
    // but Function was not forbidden. 
```

Reply:
i've left out `but Function was not forbidden` since it may or may not be forbidden

https://github.com/endojs/endo/blob/25407a9ce4bab4eaab8e87cf8288d35af4ba81a6/packages/ses/src/lockdown.js#L104-L121

(Function now tested first in its own try/catch, then eval tested separately after in its own try/catch)